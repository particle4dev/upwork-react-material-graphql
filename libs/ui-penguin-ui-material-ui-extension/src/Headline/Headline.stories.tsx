import * as React from 'react';
import IconButton from '@material-ui/core/IconButton';
import ArrowForwardIcon from '@material-ui/icons/ArrowForward';
import { withKnobs } from "@storybook/addon-knobs";

import Headline from './index';

export default {
  title: "Headline",
  decorators: [withKnobs]
};

export const Basic = () => <Headline>Basic</Headline>;

export const WithSubheader = () => <Headline subheader="subheader">WithSubheader</Headline>;

export const WithAction = () => <Headline subheader="subheader" action={
  <IconButton aria-label="forward">
    <ArrowForwardIcon />
  </IconButton>
}>WithAction</Headline>;


export const H1 = () => <Headline component="h1" subheader="subheader" action={
  <IconButton aria-label="forward">
    <ArrowForwardIcon />
  </IconButton>
}>WithAction</Headline>;
