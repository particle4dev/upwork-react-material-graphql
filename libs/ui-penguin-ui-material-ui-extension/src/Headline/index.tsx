import * as React from 'react';
import classnames from 'classnames';
import Typography from '@material-ui/core/Typography';
import { makeStyles, Theme } from '@material-ui/core/styles';

// eslint-disable-next-line @typescript-eslint/no-var-requires
const debug = require('debug')('@penguin-ui/material-ui-extension:Headline');

const useStyles = makeStyles((theme: Theme) => ({
  root: {
    display: 'flex',
    alignItems: 'center',
  },
  action: {
    flex: '0 0 auto',
    // alignSelf: 'flex-start',
  },
  content: {
    flex: '1 1 auto',
  },
}), {name: 'Headline'});

export type HeadlineProps = {
  subheader?: string,
  className?: string,
  style?: React.CSSProperties,
  component?: React.ElementType,
  action?: React.ReactNode,
  children?: React.ReactNode,
}

const Headline = React.forwardRef(function Headline(props: HeadlineProps, ref: React.Ref<HTMLDivElement>) {
  debug('render');

  const {
    children,
    subheader,
    action,
    className,
    component: Component = 'div',
    ...other
  } = props;

  const classes = useStyles({});

  return (
    <div className={classnames(classes.root, className)} ref={ref} {...other}>
      <div className={classes.content}>
        <Typography component={Component} variant="h5">{children}</Typography>
        {props.subheader && <Typography variant="body2">{subheader}</Typography>}
      </div>
      {action && <div className={classes.action}>{action}</div>}
    </div>
  );
});

if (process.env.NODE_ENV !== 'production') {
  Headline.displayName = 'penguin_ui_material_ui_extension__Headline';
}

Headline.defaultProps = {};

export default React.memo(Headline);
