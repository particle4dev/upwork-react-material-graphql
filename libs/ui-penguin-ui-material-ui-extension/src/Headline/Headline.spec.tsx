import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Headline from './index';

describe('Headline', () => {
  it('should render successfully', () => {
    const { baseElement, getByText } = render(<Headline>test</Headline>);
    expect(baseElement).toBeTruthy();
    expect(getByText('test')).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });
});
