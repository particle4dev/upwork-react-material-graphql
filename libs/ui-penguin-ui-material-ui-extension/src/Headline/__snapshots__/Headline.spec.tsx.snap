// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headline should render successfully 1`] = `
<body>
  <div>
    <div
      class="Headline-root-1"
    >
      <div
        class="Headline-content-3"
      >
        <div
          class="MuiTypography-root MuiTypography-h5"
        >
          test
        </div>
      </div>
    </div>
  </div>
</body>
`;
